libraries/lib-time-and-pitch/tests/CMakeLists.txt

```diff
@@ -1,5 +1,10 @@
 add_compile_definitions(CMAKE_SOURCE_DIR="${CMAKE_SOURCE_DIR}")
 
+# Until we discontinue support for MacOS < 10.15, we can't use <filesystem>.
+# On the other hand, recent MSVCs issue an error for using <experimental/filesystem> because deprecated.
+# We don't care here, it's just test code, let's suppress this.
+add_compile_options(-D_SILENCE_EXPERIMENTAL_FILESYSTEM_DEPRECATION_WARNING)
```

Comment:
But I would rather define something like "USE_EXPERIMENTAL_FILESYSTEM" only on macOS to avoid the potential compilation breakdown when some newer compiler decides to drop this headers from the stdlib